umqtt.simple/umqtt/simple.py

```diff
@@ -43,9 +45,15 @@ def connect(self, clean_session=True):
         msg = bytearray(b"\x10\0\0\x04MQTT\x04\x02\0\0")
         msg[1] = 10 + 2 + len(self.client_id)
         msg[9] = clean_session << 1
+        if self.user and self.pswd:
```

Comment:
I'd say it's an error to specify user without pswd, and vice versa.  If you specify one you should specify both.  Thus, it should be enough to check that user is not None (ie `if self.user is not None:`).